src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/electric/machines/enchanting/AbstractEnchantmentMachine.java

```diff
@@ -55,4 +61,23 @@ protected void showEnchantmentLevelWarning(@Nonnull BlockMenu menu) {
         menu.replaceExistingItem(22, progressBar);
     }
 
+    protected boolean hasIgnoredLore(@Nonnull ItemStack itemStack) {
+        List<String> ignoredLore = ignoredLores.getValue();
+        // Skip the check if not set any ignored lore.
+        if (ignoredLore.isEmpty()) {
+            return false;
+        }
+        if (itemStack.hasItemMeta()) {
+            List<String> itemLore = itemStack.getItemMeta().getLore();
+            if (itemLore == null) {
```

Comment:
as Cookie mentioned before, probably best to do hasLore, quick circuit-break if empty too (and if they change stuff in the future it'll be supported)

Reply:
I realized that I forgot to press "submit" on my review

Reply:
OK. I thought empty list are rare to see so didn't check.